File: src/main/java/com/freya02/docs/data/TargetType.java
package com.freya02.docs.data;

import org.jetbrains.annotations.Nullable;

public enum TargetType {
	CLASS,
	METHOD,
	FIELD,
	UNKNOWN;

	public static TargetType fromFragment(@Nullable String fragment) {
		if (fragment == null) {
			return CLASS;
		} else if (fragment.contains("(")) {
			return METHOD;
		} else {
			return FIELD;
		}
	}
}

File: src/main/java/com/freya02/docs/data/TargetType.kt
package com.freya02.docs.data

enum class TargetType(val id: Int) {
    CLASS(1),
    METHOD(2),
    FIELD(3),
    UNKNOWN(-1);

    companion object {
        @JvmStatic
        fun fromFragment(fragment: String?): TargetType {
            return when {
                fragment == null -> CLASS
                fragment.contains("(") -> METHOD
                else -> FIELD
            }
        }
    }
}